src/http/one/TeChunkedParser.cc

```diff
@@ -115,55 +116,66 @@ Http::One::TeChunkedParser::parseChunkSize(Http1::Tokenizer &tok)
  * ICAP 'use-original-body=N' extension is supported.
  */
 bool
-Http::One::TeChunkedParser::parseChunkExtension(Http1::Tokenizer &tok, bool skipKnown)
+Http::One::TeChunkedParser::parseChunkExtensions(Tokenizer &tok, const bool skipKnown)
 {
+    while (parseChunkExtension(tok, skipKnown)) {
+        buf_ = tok.remaining(); // got one extension; there may be more
+    }
+
+    if (skipLineTerminatorIfAny(tok)) {
+        buf_ = tok.remaining();
+        parsingStage_ = theChunkSize ? Http1::HTTP_PARSE_CHUNK : Http1::HTTP_PARSE_MIME;
+        return true; // reached the end of extensions (if any)
+    }
+
+    static const CharacterSet NonLF = (CharacterSet::LF).complement().rename("non-LF");
+    if (tok.skipAll(NonLF) && tok.skip('\n'))
+        throw TexcHere("cannot parse chunk extension"); // <garbage> CR*LF
+    return false; // need more data to finish parsing the extension
```

Comment:
Drop the incorrect "to finish parsing the extension" -- we may need more data to finish parsing CRLF:

```suggestion
    return false; // need more data
```